src/publicodes-state/helpers/getSortedQuestionsList.ts

```diff
@@ -0,0 +1,54 @@
+import { DottedName } from '../types'
+
+type Props = {
+  questions: DottedName[]
+  categories: string[]
+  subcategories: Record<string, string[]>
+  missingVariables: Record<string, number>
+}
+
+export default function getSortedQuestionsList({
+  questions,
+  categories,
+  subcategories,
+  missingVariables,
+}: Props): DottedName[] {
+  return questions.sort((a, b) => {
+    const aSplittedName = a.split(' . ')
+    const bSplittedName = b.split(' . ')
+
+    // We first sort by category
+    if (
+      categories.indexOf(aSplittedName[0]) >
+      categories.indexOf(bSplittedName[0])
+    ) {
+      return 1
+    }
+    if (
+      categories.indexOf(aSplittedName[0]) <
+      categories.indexOf(bSplittedName[0])
+    ) {
+      return -1
+    }
+
+    // then by subcategory
+    const categoryOfBothQuestions = aSplittedName[0]
+    const aCategoryAndSubcategory = aSplittedName[0] + ' . ' + aSplittedName[1]
+    const bCategoryAndSubcategory = bSplittedName[0] + ' . ' + bSplittedName[1]
+    if (
+      subcategories[categoryOfBothQuestions].indexOf(aCategoryAndSubcategory) >
+      subcategories[categoryOfBothQuestions].indexOf(bCategoryAndSubcategory)
+    ) {
+      return 1
+    }
+    if (
+      subcategories[categoryOfBothQuestions].indexOf(aCategoryAndSubcategory) <
+      subcategories[categoryOfBothQuestions].indexOf(bCategoryAndSubcategory)
+    ) {
+      return -1
+    }
+
+    // then by missing variables score
+    return missingVariables[b] - missingVariables[a]
```

Comment:
Je comprends pas bien, ce bout de code n'est jamais accessible si ? tes deux ifs à partir de la ligne 20 couvrent 100% des cas

Reply:
Les 2 ifs ne couvrent pas tous les cas, ici on gère l'ordre à l'intérieur des sous catégories

Reply:
Suite à notre discussion rapide hier, il me semble qu'on va bien au bout du "sort" grâce à l'inégalité stricte non ?

Si c'est même catégorie, même sous catégorie, on arrive bien aux missing

Reply:
Han okay ty as sans doute raison !